javascript/test/test_refactor.js

```diff
@@ -35,22 +35,37 @@ describe('refactor', () => {
             assert(ast.body[0].expression.createdByPiranha !== undefined);
         });
 
-        it('should replace calls to flag check APIs with boolean literals', () => {
-            const code = `if (isFlagTreated(testFlag)) {} else {}`;
-            const expected_code = `if (true) {} else {}`;
-            const ast = recast.parse(code).program;
-
-            const engine = new refactor.RefactorEngine(ast, properties, treated, flagname);
-            engine.flagAPIToLiteral();
-            engine.finalizeLiterals();
-
-            const refactored_code = recast.print(ast).code;
-
-            assert(engine.changed);
-            assert(
-                expected_code === refactored_code,
-                `\nEXPECTED : ${JSON.stringify(expected_code)}\nREFACTORED : ${JSON.stringify(refactored_code)}`,
-            );
+        const booleanLiteralReplacementSpecs = [
+            {
+                name: 'Identifier',
+                code: `if (isFlagTreated(testFlag)) {} else {}`,
+                expectedCode: 'if (true) {} else {}',
+            },
+            {
+                name: 'StringLiteral',
+                code: `if (isFlagTreated('testFlag')) {} else {}`,
+                expectedCode: 'if (true) {} else {}',
+            },
+        ];
+
+        booleanLiteralReplacementSpecs.forEach((spec) => {
+            it(`should replace ${spec.name} calls to flag check APIs with boolean literals`, () => {
+                const ast = recast.parse(spec.code).program;
+
+                const engine = new refactor.RefactorEngine(ast, properties, treated, flagname);
+                engine.flagAPIToLiteral();
+                engine.finalizeLiterals();
+
+                const refactoredCode = recast.print(ast).code;
+
+                assert(engine.changed);
+                assert(
+                    spec.expectedCode === refactoredCode,
+                    `\nEXPECTED : ${JSON.stringify(spec.expected_code)}\nREFACTORED : ${JSON.stringify(
```

Comment:
Shouldn't this be `spec.expectedCode` instead of `spec.expected_code`? Actually, I think it's better if we use underscore casing throughout (i.e. rename `spec.expectedCode` to `spec.expected_code` everywhere) since that's what has been used in the rest of the tests.

Reply:
Good catch - I'm not sure how this test passed locally...

My JS instincts kicked and I unconsciously wrote this camel case -- I'll comb through this PR and make sure it's only using snake case.

Reply:
This would have been a problem only if the refactoring failed. Since the refactoring is correct, the tests passed. 
Impressive that this was caught in code review 💯

Reply:
🤦  Yup, that typo is in the assertion failure message. Thanks all